gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.go-license-file-generator.gradle.kts

```
@@ -0,0 +1,149 @@
+            val licenseResourcesForBinary = resourceLicenseDir.dir("$it-licenses").asFile
+            if (!areDirectoriesEqual(generatedLicenses, licenseResourcesForBinary, logger)) {
+                val message = """
+                [FAILURE] License file validation failed!
+                Generated license files differ from committed files at $resourceLicenseDir.
+                To update the committed license files, run './gradlew generateLicenseResources' and commit the changes.
+
+                Note: This will completely regenerate all license files under $resourceLicenseDir and remove any stale ones.
+                """
+                throw GradleException(message)
+            } else {
+                logger.lifecycle("Go license file validation succeeded: Generated go license files match the committed ones.")
+            }
+        }
+    }
+}
+
+tasks.named("validateLicenseFiles") {
+    dependsOn(validateGoLicenses)
+}
+
+/**
+ * This tasks requires that the go-license generation is already done
+ * during 'goLicenseGenerationConfig.generatingGoLicensesGradleTask.' task.
+ */
+val generateGoLicenses = tasks.register("generateGoLicenses") {
+    // Generating the licenses with go-license tool is done during 'dockerCompileGo' task
+    dependsOn(goLicenseGenerationConfig.generatingGoLicensesGradleTask.get())
+    doLast {
+        transformGoLicenseOutputToDesiredLicenseStructure(goLicenseGenerationConfig.buildGoLicenseFilesDir.get(), logger)
+
+        // copy included license
+        val newIncludedLicensePath = goLicenseGenerationConfig.buildGoLicenseFilesDir.get().resolve("resources/LICENSE.txt").toPath()
+        Files.copy(goLicenseGenerationConfig.binaryLicenseFile.get().toPath(), newIncludedLicensePath, StandardCopyOption.REPLACE_EXISTING)
+    }
+}
+
+val generateGoLicenseResources = tasks.register("generateGoLicenseResources") {
+    description = "Copies generated license files to the resources directory"
+    dependsOn(generateGoLicenses)
+
+    doLast {
+        val generatedLicenses = goLicenseGenerationConfig.buildGoLicenseFilesDir.get().resolve("resources")
+        goLicenseGenerationConfig.packagedBinaries.get().forEach {
+            val licenseResourcesForBinary = resourceLicenseDir.dir("$it-licenses").asFile
+
+            Files.createDirectories(licenseResourcesForBinary.toPath())
+            copyDirectory(generatedLicenses, licenseResourcesForBinary, logger)
+        }
+    }
+}
+
+tasks.named("generateLicenseResources") {
+    dependsOn(generateGoLicenseResources)
+}
+
+fun transformGoLicenseOutputToDesiredLicenseStructure(
+    licenseFolder: File,
+    logger: Logger,
+) {
+    val generatedDirectory = licenseFolder.resolve("generated").toPath()
+    val transformedDirectory = licenseFolder.resolve("resources/THIRD_PARTY_LICENSES").toPath()
+
+    // 1. Validation: Ensure input exists
+    if (!Files.exists(generatedDirectory) || !Files.isDirectory(generatedDirectory)) {
```

The `isDirectory()` already checks if it exist: https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#isDirectory-java.nio.file.Path-java.nio.file.LinkOption...-

```suggestion
    if (!Files.isDirectory(generatedDirectory)) {
```